### PR TITLE
Add golden tests for demo page

### PR DESCRIPTION
**Pull request by perclasson**
Closes https://github.com/flutter/gallery/issues/39

### CODE REVIEW

File: golden_test/testing/font_loader.dart [hunk 1]
```diff
@@ -35,10 +36,29 @@ Future<Map<String, List<Future<ByteData>>>> loadFontsFromManifest() async {
   return fontFamilyToData;
 }
 
+Map<String, List<Future<ByteData>>> loadFontsFromTestingDir() {
```
Comment (rami-a) [inline golden_test/testing/font_loader.dart:39]: Why do all the roboto font files have to be included now?

File: golden_test/demo_test.dart [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:gallery/main.dart';
+
+import 'testing/precache_images.dart';
+import 'testing/util.dart';
+
+const demoBannerRoute = '/demo/banner';
+
+void main() {
```
Comment (rami-a) [inline golden_test/demo_test.dart:15]: As discussed offline, rename all these tests to not specify banner but instead be more generic for demo page

File: golden_test/testing/font_loader.dart (hunk 1 above)
Comment (perclasson) [reply golden_test/testing/font_loader.dart:39]: We were not using the Roboto font on the home page, but it is on the demo page, so I never noticed that we needed to load it as well.

In the previous code we were only loading the google fonts that were added and the icons from the `pubspec.yaml`. As the Roboto font is added by default in Flutter and not added in the `pubspec.yaml` or in the `google_fonts` folder, we need to explicitly load it here in the tests.

File: golden_test/demo_test.dart (hunk 2 above)
Comment (perclasson) [reply golden_test/demo_test.dart:17]: Done!

**Pull request merged**